src/gui/PatternsManagerWindow.hpp

```diff
@@ -29,14 +47,19 @@ class PatternsManagerWindow final : public QDialog
 
     ~PatternsManagerWindow() override; // = default
 
-    void update_patterns(QComboBox &box);
+    void update_patterns(QListWidget &list);
 
 private:
     std::unique_ptr<Ui::PatternsManagerWindow> ui_;
     Settings &settings_; // NOLINT(cppcoreguidelines-avoid-const-or-ref-data-members)
 
+    void update_pattern(QListWidgetItem *item);
+    void move_pattern(const QModelIndex &sourceParent,
+                      int sourceStart,
+                      int sourceEnd,
+                      const QModelIndex &destinationParent,
+                      int destinationRow);
```

Comment:
Please use snake_case

Reply:
Fixed, also made the unused parameters anonymous since they are only there to conform to the callback specification. Is that okay, or would `[[maybe_unused]]` be better?

Reply:
I'm failing to compile this on Linux (compile error in bethutil: `compression_device.cpp:170:84: error: ‘struct btu::tex::detail::DxAdapter’ has no member named ‘gpu_name’`)

But I think you are probably not using clang tidy. There are several lints it should have recommended
In this case, I think we are supposed to put unused parameters like this: `(int /* parameterName */)`

Reply:
Hmm, that is very weird since I haven't directly touched bethutil in this PR. 🤔
I have been building CAO against my local bethutil fork, so there is that. Checking what version is pinned in the port.

Reply:
`fatal error C1083: Cannot open include file: 'compression_device.hpp': No such file or directory` weirdly enough.

Reply:
Regarding compilation - can you try clearing your cmake cache? That helped. But it seems something is fishy with `compression_device.cpp/hpp`. I don't think I have touched that one at all.

Reply:
Regarding clang tidy - so far I haven't been able to make it work with Visual Studio 2022.